main/background.ts

```diff
@@ -62,7 +62,36 @@ function createGraphqlSeverProcess() {
   });
 }
 
-app.on("ready", () => {
+async function isGraphQLServerReady(url: string): Promise<boolean> {
+  try {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ query: "{ __typename }" }),
+    });
+    console.log(url, response);
+    return response.ok;
+  } catch (error) {
+    console.error("GraphQL server is not ready:", error);
+    return false;
+  }
+}
+
+async function waitForGraphQLServer(
+  url: string,
+  timeout: number = 30000,
+): Promise<void> {
+  const startTime = Date.now();
+  while (Date.now() - startTime < timeout) {
+    if (await isGraphQLServerReady(url)) {
+      return;
+    }
```

Comment:
Does the below work? I think it's bad practice to put an `await` in an if block like that
```suggestion
    const isReady = await isGraphQLServerReady(url)
    if (isReady) {
      return;
    }
```